### PR TITLE
Include a space before single quote and bang when pretty-printing.

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes issue 782.

### CODE REVIEW

**Comment by alexeagle** [comment]
Thanks, I looked through the unary operators but wasn't thinking about how they can appear prefix or suffix, and that makes a difference. Added the missing test cases and will find a fix.

**Comment by alexeagle** [comment]
Ping!

File: src/com/google/javascript/jscomp/CodeGenerator.java [hunk 1]
```diff
@@ -198,6 +198,7 @@ void add(Node n, Context context) {
       case Token.RETURN:
         add("return");
         if (childCount == 1) {
+          cc.maybeInsertSpace();
```
Comment (blickly) [inline src/com/google/javascript/jscomp/CodeGenerator.java]: Let's also do the same for yield and throw, while we're here.

Comment (alexeagle) [inline src/com/google/javascript/jscomp/CodeGenerator.java]: Done.

File: test/com/google/javascript/jscomp/CodePrinterTest.java [hunk 2]
```diff
@@ -978,6 +967,50 @@ public void testPrettyPrinter_defaultValue() throws Exception {
     assertPrettyPrint("(a=1)=>123;", "(a = 1) => 123;\n");
   }
 
+  // For https://github.com/google/closure-compiler/issues/782
+  public void testPrettyPrinter_spaceBeforeSingleQuote() throws Exception {
+    assertPrettyPrint("var f = function() { return 'hello';};",
```
Comment (dimvar) [inline test/com/google/javascript/jscomp/CodePrinterTest.java]: Was the missing space before } intentional in this test?

**Comment by dimvar** [comment]
OK, looks good. I just had a minor comment. Can you resolve the conflicts and squash the commits? And then I'll merge it.

File: test/com/google/javascript/jscomp/CodePrinterTest.java (hunk 2 above)
Comment (alexeagle) [inline test/com/google/javascript/jscomp/CodePrinterTest.java]: No, it's inconsequential. Added a space.

**Comment by alexeagle** [comment]
All set.

**Pull request merged**